src/internal_telemetry/allocations/mod.rs

```diff
@@ -15,14 +59,45 @@ pub struct MainTracer;
 
 impl Tracer for MainTracer {
     #[inline(always)]
-    fn trace_allocation(&self, _object_size: usize, _group_id: AllocationGroupId) {}
+    fn trace_allocation(&self, object_size: usize, group_id: AllocationGroupId) {
+        GROUP_MEM_ALLOCS[group_id.as_raw()].fetch_add(object_size as u64, Ordering::Relaxed);
+    }
 
     #[inline(always)]
-    fn trace_deallocation(&self, _object_size: usize, _source_group_id: AllocationGroupId) {}
+    fn trace_deallocation(&self, object_size: usize, source_group_id: AllocationGroupId) {
+        GROUP_MEM_DEALLOCS[source_group_id.as_raw()]
+            .fetch_add(object_size as u64, Ordering::Relaxed);
+    }
 }
 
 /// Initializes allocation tracing.
-pub const fn init_allocation_tracing() {}
+pub fn init_allocation_tracing() {
+    let alloc_processor = thread::Builder::new().name("vector-alloc-processor".to_string());
+    alloc_processor
+        .spawn(|| {
+            without_allocation_tracing(|| loop {
+                for idx in 0..GROUP_MEM_ALLOCS.len() {
+                    let allocs = GROUP_MEM_ALLOCS[idx].load(Ordering::Relaxed);
+                    let deallocs = GROUP_MEM_DEALLOCS[idx].load(Ordering::Relaxed);
+                    let mem_used = allocs - deallocs;
+
+                    if allocs == 0 {
+                        continue;
+                    }
+                    let group_info = GROUP_INFO[idx].lock().unwrap();
+                    gauge!(
+                        "component_bytes_allocated",
```

Comment:
```suggestion
                        "component_allocated_bytes",
```

This follows closer to our naming conventions which puts the units last.

Reply:
Oof, yeah, that's my bad. I suggested `component_bytes_allocated` to better match our existing metrics.. but had an incorrect vision in my mind of it being noun/verb instead of verb/noun.